candidate_base.go

```diff
@@ -231,6 +233,21 @@ func (c *candidateBase) recvLoop(initializedCh <-chan struct{}) {
 	}
 }
 
+func (c *candidateBase) validateSTUNTrafficCache(addr net.Addr) bool {
+	if candidate, ok := c.remoteCandidateCaches[toAddrPort(addr)]; ok {
```

Comment:
A we sure `validateSTUNTrafficCache()` is not invoked concurrently?

Reply:
If `start function of candidateBase` was not atomic, It should be not thread safe.

I thinks it is fine because `addCandidate function of Agent` is checking that candidateBase start is atomic.
and Candidate always created before called `addCandidate`.

Reply:
Ah I see. Thanks for the clarification.

I was somehow under the impression that we have a cache shared by all candidates. But that does not make sense at all..